packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol

```diff
@@ -524,6 +530,25 @@ contract PerpetualPositionManager is FundingRateApplier {
         return;
     }
 
+    /**
+     * @notice Drains any excess balance of the provided ERC20 token to a pre-selected beneficiary.
+     * @dev This will drain down to the amount of tracked collateral and drain the full balance of any other token.
+     * @param token address of the ERC20 token whose excess balance should be drained.
+     */
+    function trimExcess(IERC20 token) external fees() nonReentrant() returns (FixedPoint.Unsigned memory amount) {
+        FixedPoint.Unsigned memory balance = FixedPoint.Unsigned(token.balanceOf(address(this)));
+
+        if (address(token) == address(collateralCurrency)) {
+            // If it is the collateral currency, send only the amount that the contract is not tracking.
+            // Note: this could be due to rounding error or balance-changing tokens, like aTokens.
+            amount = balance.sub(_pfc());
```

Comment:
is it the case that pfc is essentially all the collateral held by sponsors?

Reply:
Correct